src/main/java/frc/robot/subsystems/Hood.java

```diff
@@ -40,6 +46,29 @@ public double getHoodMotorEncoderCount() {
   public void periodic() {
     // This method will be called once per scheduler run
     SmartDashboard.putNumber("Hood Motor", getHoodMotorEncoderCount());
+  }
+
+  public void setSpeed(double p_speed) {
+
+    // If limit switch is pressed, then stop
+    if (hoodSwitch.get()) {
+      setSpeed(0);
+      resetEncoderCounts();
+    }
+
+    // If over 90 degrees, then stop
+    if (getHoodAngle() >= 90 && p_speed >= 0) {
+      setSpeed(0);
+    }
+
+    // If under 0 degrees, then stop
+    if (getHoodAngle() <= 0 && p_speed <= 0) {
+      setSpeed(0);
+    }
+    hoodMotor.set(ControlMode.PercentOutput, p_speed);
```

Comment:
Check your safety logic. It's not doing what you think. Also you are doing recursion, which you do not want.